munit/shared/src/main/scala/munit/MUnitRunner.scala

```diff
@@ -465,4 +455,13 @@ object MUnitRunner {
       case nsme: NoSuchMethodException => false
     }
   }
+
+  // NOTE(olafur): these exceptions appear when we await on futures. We unwrap
+  // these exception in order to provide more helpful error messages.
+  private[munit] def rootCause(x: Throwable): Throwable = x match {
```

Comment:
We should add a note here to preserve bincompat since this is intended to be used by integrations.